### PR TITLE
initial attempt

### PR DESCRIPTION
**Pull request by otuffaha**

### CODE REVIEW

File: src/pages/recipes/components/RecipeDetailsPage.tsx [hunk 1]
```diff
@@ -29,14 +29,18 @@ const CalorieLabel = styled(Label)`
...
+  unlikeRecipe: () => void;
+  getLikedRecipes: () => void;
   onShare?: () => void;
   onBack: () => void;
   recipeDetails: RecipeDetails;
 };
 
 const RecipeDetailsPage: React.FC<Props> = ({
-  onFavorite,
+  likeRecipe,
```
Comment (mavalos1) [inline src/pages/recipes/components/RecipeDetailsPage.tsx]: likeRecipe likely needs a recipeId to like right?

File: src/pages/recipes/components/RecipeDetailsPage.tsx [hunk 2]
```diff
@@ -29,14 +29,18 @@ const CalorieLabel = styled(Label)`
 `;
 
 export type Props = {
-  onFavorite?: () => void;
+  likeRecipe: () => void;
+  unlikeRecipe: () => void;
```
Comment (mavalos1) [inline src/pages/recipes/components/RecipeDetailsPage.tsx]: same with unlikeRecipe

File: src/pages/recipes/components/RecipeDetailsPage.tsx [hunk 3]
```diff
@@ -29,14 +29,18 @@ const CalorieLabel = styled(Label)`
...
+  getLikedRecipes: () => void;
```
Comment (mavalos1) [inline src/pages/recipes/components/RecipeDetailsPage.tsx]: getLikedRecipes is not used here but in list view, aka RecipeListContainer

File: src/pages/recipes/containers/RecipeDetailsContainer.tsx [hunk 4]
```diff
@@ -39,12 +42,16 @@ class RecipeDetailsContainer extends React.Component<Props> {
 
 const mapStateToProps = (state: any) => ({
   recipeDetails: state.recipe.recipeDetails,
+  onFavorite: state.likeRecipe,
```
Comment (mavalos1) [inline src/pages/recipes/containers/RecipeDetailsContainer.tsx]: onFavorite is the likeRecipe function, you should pass a favorited boolean prop instead

**Review by mavalos1 — CHANGES_REQUESTED** [review]

File: src/pages/recipes/containers/RecipeDetailsContainer.tsx [hunk 5]
```diff
@@ -32,6 +34,9 @@ class RecipeDetailsContainer extends React.Component<Props> {
           navigation.pop(1);
         } }
         recipeDetails={ recipeDetails }
+        likeRecipe = { likeRecipe }
+        unlikeRecipe = { unlikeRecipe }
+        favorited = {// how to get this val}
```
Comment (mavalos1) [inline src/pages/recipes/containers/RecipeDetailsContainer.tsx]: should a field in the Recipe when fetch, isn't it? you did added it to the data model on backend

Comment (mavalos1) [inline src/pages/recipes/containers/RecipeDetailsContainer.tsx]: and no-op but this unended curly brace will crash the app

File: src/pages/recipes/components/RecipeDetailsPage.tsx [hunk 6]
```diff
@@ -59,7 +72,7 @@ const RecipeDetailsPage: React.FC<Props> = ({
     </Box>
     <DrawerCard
       topRightOverlay={ (
-        <IconButton onPress={ onFavorite }>
+        <IconButton onPress={ () => onFavorite(state.favorited) }>
```
Comment (mavalos1) [inline src/pages/recipes/components/RecipeDetailsPage.tsx]: prefer destructuring i thought I added an eslint rule for this

File: src/pages/recipes/containers/RecipeListContainer.tsx [hunk 7]
```diff
@@ -10,19 +10,23 @@ import { Recipe } from '../../../constants/dataTypes';
...
   recipes: Array<Recipe>;
   getRecipesByUser: (id: number) => void;
+  getLikedRecipes: () => Array<Recipe>;
 };
 
 class RecipeListPage extends React.Component<Props> {
   public componentDidMount(): void {
     const { userId, getRecipesByUser } = this.props;
     getRecipesByUser(userId);
+    getLikedRecipes();
```
Comment (mavalos1) [inline src/pages/recipes/containers/RecipeListContainer.tsx]: and after getting it you need to hook it up to the list as well

File: src/pages/recipes/containers/RecipeDetailsContainer.tsx (hunk 5 above)
Comment (otuffaha) [reply src/pages/recipes/containers/RecipeDetailsContainer.tsx]: i don't think so

Comment (mavalos1) [reply src/pages/recipes/containers/RecipeDetailsContainer.tsx]: then how did the favorite work on the backend, what kind of flags are added?

**Review by mavalos1 — APPROVED** [review]

**Pull request merged**